src/daemon/emitter.cc

```diff
@@ -394,18 +386,7 @@ void Emitter::DoRemoteExecute(const base::WorkerPool& pool,
     auto& source = std::get<SOURCE>(*task);
     auto& extra_files = std::get<EXTRA_FILES>(*task);
 
-    // Check that we have a compiler of a requested version.
-    net::proto::Status status;
-    if (!SetupCompiler(incoming->mutable_flags(), &status)) {
-      std::get<CONNECTION>(*task)->ReportStatus(status);
-      continue;
-    }
-
-    auto outgoing = std::make_unique<proto::Remote>();
-    if (source.str.empty() && !GenerateSource(incoming, &source)) {
-      failed_tasks_->Push(std::move(*task));
-      continue;
-    }
+    CHECK(!source.str.empty());
```

Comment:
It should be here in any case. To put to `all_tasks_` we need to now hash of sources of the task. All these fields are filled in `PopulateTask`. There should be no way to put a task to queue without knowing it's hash/sources.